### PR TITLE
More informations about migrating from 1.6

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | few important points about module migration
| Fixed ticket? | n/a
| Sponsor company | impSolutions

### CODE REVIEW

File: src/content/1.7/modules/core-updates/1.6.md [hunk 1]
```diff
@@ -39,3 +39,39 @@ Media::addJsDef([
 +    'translations' => $translations, // Data will be encoded by PrestaShop
 ]);
 ```
+
+## Module front controllers
+
+There are serveral changes to templating in your module front controllers.
```
Comment (matks) [inline src/content/1.7/modules/core-updates/1.6.md:45]:
```suggestion
There are several changes to templating in your module front controllers.
```

**Review by matks — COMMENTED** [review]
Hi @kpodemski this page currently contains one big "Backward compatibility with PrestaShop 1.6" section which aims to explain how to take a module compatible with PS 1.7 and add compatibility for PS 1.6

If I am not wrong your PR adds some hints about "how to take a module compatible with PS 1.6 and add compatibility for PS 1.7"

So maybe we should create a 2nd section in this page 😉 rather than putting the new content at the end of it.

One section "how to take a module compatible with PS 1.7 and add compatibility for PS 1.6" and one section for the opposite

**Comment by kpodemski** [comment]
@matks 
Hmmm, I thought about this place because without it modules doesn't work, completely, but you're right, I've missed a point of this page, it's about making module compatible with 1.6, not 1.7 🤦 

So... you're suggesting:

- to change page title to `Modules compatibility`?
- and split it into:
-- Make PrestaShop 1.6 modules compatible with 1.7
-- Make PrestaShop 1.7 modules compatible with 1.6

Am I understanding you correctly?

Thanks

**Comment by matks** [comment]
I suggest to keep the title of the page (for SEO purposes) but to split it the 2 sections as you said 😄

**Review by Progi1984 — APPROVED** [review]

File: src/content/1.7/modules/core-updates/1.6.md [hunk 2]
```diff
@@ -39,3 +39,38 @@ Media::addJsDef([
...
 ]);
 ```
+
+## Module front controllers
+
+There are several changes to templating in your module front controllers.
+
+### Getting a proper template for a page
+
+In order to display content properly in your `ModuleFrontController` files, you need to change a way you link to your template file in your `initContent` method:
```
Comment (matthieu-rolland) [inline src/content/1.7/modules/core-updates/1.6.md:49]:
```suggestion
In order to display content properly in your `ModuleFrontController` files, you need to change the way you link to your template file in your `initContent` method:
```

File: src/content/1.7/modules/core-updates/1.6.md [hunk 3]
```diff
@@ -39,3 +39,38 @@ Media::addJsDef([
...
+
+
+```patch
+- $this->setTemplate('views/templates/front/template.tpl');
++ $this->setTemplate('module:yourmodule/views/templates/front/template.tpl');
+```
+
+### Basic structure of a module front page template
+
+If you want your module to display entire page with your own content, basic structure of your `.tpl` looks like this:
```
Comment (matthieu-rolland) [inline src/content/1.7/modules/core-updates/1.6.md:59]:
```suggestion
If you want your module to display an entire page with your own content, the basic structure of your `.tpl` file looks like this:
```

File: src/content/1.7/modules/core-updates/1.6.md [hunk 4]
```diff
@@ -39,3 +39,38 @@ Media::addJsDef([
...
+{/block}
+
+{block name='page_content'}
+Content of the page
+{/block}
+```
+
+[Template inheritance](https://devdocs.prestashop.com/1.7/themes/reference/template-inheritance/) is powerful mechanism available from v1.7 of PrestaShop.
+
+More informations about front office and templating available [here](https://devdocs.prestashop.com/1.7/modules/creation/displaying-content-in-front-office/#embedding-a-template-in-the-theme) and [here](https://devdocs.prestashop.com/1.7/themes/).
```
Comment (matthieu-rolland) [inline src/content/1.7/modules/core-updates/1.6.md:75]:
```suggestion
More informations about front office and templating are available [here](https://devdocs.prestashop.com/1.7/modules/creation/displaying-content-in-front-office/#embedding-a-template-in-the-theme) and [here](https://devdocs.prestashop.com/1.7/themes/).
```

**Review by matthieu-rolland — COMMENTED** [review]
some minor feedback

**Comment by sowbiba** [comment]
LGTM, waiting for you to handle @matthieu-rolland feedbacks

**Comment by kpodemski** [comment]
Hey @matks 

I've decided to separate this page, this is in my opinion an important part for people still migrating their modules from 1.6 to 1.7, let me know what do you think

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
@kpodemski Ready to review ?

**Comment by kpodemski** [comment]
@Progi1984 yep

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @kpodemski

**Pull request merged**